### PR TITLE
MGMT-15359: create a virtual device for coreos-installer

### PR DESCRIPTION
**Pull request by danielerez**
We need a solution for the limitation in coreos-installer that can't install on a device with mounted partitions.
Currently, we are loading the rootfs image to memory on boot, thus, no partitions are mounted on installation device. However, this workaround works only with UEFI when secure boot an TPM are disabled. Also, it isn't supported for some baremetals when using an older GRUB version.

As an alternative solution to mitigate those issues, we are now creating a multipath virtual device instead. As we build the device without the mounted partitions, it can be passed to coreos-installer and the installation is successfully completed.

The following changes were made to support this solution:
* Download raw variant of CoreOS base image (instead of qcow2)
  * Required for using go-diskfs to fetch partitions info
* Added create-virtual-device.sh script for generating the partition table and build the device.
  * Using partitions start-sector and size fetched with go-diskfs.
* Removed old workaround of overriding the EFI folder.

Note: this change depends on supporting the multipath device in assisted-installer-agent.

### CODE REVIEW

File: data/scripts/bin/create-virtual-device.sh.template [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Configure a loop device for boot.raw
+losetup /dev/loop2 /tmp/boot.raw
+
+# Create a virtual device for passing to coreos-installer
+dmsetup create agent <<.
+0 2048 linear /dev/loop2 0
+{{.Partition0.StartSector}} {{.Partition0.Size}} linear /dev/sda1  0
+{{.Partition1.StartSector}} {{.Partition1.Size}} linear /dev/sda2  0
+{{.Partition2.StartSector}} {{.Partition2.Size}} linear /dev/sda3  0
+{{.Partition3.StartSector}} {{.Partition3.Size}} linear /dev/sda4  0
```
Comment (jhernand) [inline data/scripts/bin/create-virtual-device.sh.template:13]: Will these partition devices always be `/dev/sda...`? What if they are `/dev/nvme0n1p...`?

Comment (danielerez) [reply data/scripts/bin/create-virtual-device.sh.template:13]: Good point. Maybe we can extract the boot device first with something like:
```
fdisk -l | grep dev | grep boot | cut -f1 -d'1'
```

Comment (jhernand) [reply data/scripts/bin/create-virtual-device.sh.template:13]: If I understood correctly this script is running before ignition, right? If so maybe you can use `/dev/disk/by-id/coreos-boot-disk`. Should be a link to the actual boot device. See https://docs.fedoraproject.org/en-US/fedora-coreos/storage/#_referencing_block_devices_from_ignition.

Comment (danielerez) [reply data/scripts/bin/create-virtual-device.sh.template:13]: > /dev/disk/by-id/coreos-boot-disk

Actually this scripts runs after the ignition. I.e. we include the script as part of the ignition, but it runs only after the machine is completely booted.

Comment (danielerez) [reply data/scripts/bin/create-virtual-device.sh.template:13]: This also works:
```
lsblk -rfp | grep agentboot | cut -f1 -d '5'
```

Comment (jhernand) [reply data/scripts/bin/create-virtual-device.sh.template:13]: The maybe use `/dev/disk/by-partlabel/...`?

Comment (danielerez) [reply data/scripts/bin/create-virtual-device.sh.template:13]: > Then maybe use `/dev/disk/by-partlabel/...`?

You mean something like:
```
{{.Partition0.StartSector}} {{.Partition0.Size}} linear /dev/disk/by-partlabel/BIOS-BOOT  0
{{.Partition1.StartSector}} {{.Partition1.Size}} linear /dev/disk/by-partlabel/EFI-SYSTEM 0
{{.Partition2.StartSector}} {{.Partition2.Size}} linear /dev/disk/by-partlabel/boot  0
{{.Partition3.StartSector}} {{.Partition3.Size}} linear /dev/disk/by-partlabel/root 0
```

Should be fine. But is it unique? I mean if we have another disk with the same partition naming?

Comment (jhernand) [reply data/scripts/bin/create-virtual-device.sh.template:13]: Yes, I mean that.

It is not unique, you can potentially have multiple partitions with the same label, but that would be a broken system. I don't think that is an issue for our use case.

**Review by jhernand — APPROVED** [review]
Looks good to me.

**Comment by danielerez** [comment]
> New changes are detected. LGTM label has been removed.

fixed lint.

**Review by jhernand — APPROVED** [review]

**Pull request merged**